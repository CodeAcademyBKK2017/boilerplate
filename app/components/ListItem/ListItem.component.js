import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import style from './ListItem.component.style';
import Swipeout from 'react-native-swipeout';
import {
  FlatList,
  Text,
  TouchableOpacity,
  View
} from 'react-native';

export default class ListItem extends Component {

  _keyExtractor = (item) => item.id;

  renderItemList = ({item}) => {
    const swipeoutBtns = [
      {
        text: 'Delete',
        type: 'delete',
        onPress: this.props.onDelete(item)
      }
    ];

    return <Swipeout backgroundColor= '#F5F5F5' right={swipeoutBtns}>
      <TouchableOpacity 
        onPress={ this.props.onShowModal(item)}>
        <View style={style.marginStyle}>
          <View style={style.flexStyle}>
            <Text style={style.textTitleStyle}>{item.title}</Text>
            <Text style={style.textContentStyle}>{item.content}</Text>
          </View>
        </View>
        <View style={style.divide}/>
      </TouchableOpacity>
    </Swipeout>;
  }
  render () {
    return (
      <View>
        <Text style={style.textNotesStyle}>Notes:</Text>
        <FlatList 
          style={style.flatListStyle}
          data={this.props.dataNotes}
          keyExtractor={this._keyExtractor}
          renderItem={this.renderItemList}/>
      </View>
    );
  }
}

ListItem.propTypes = {
  dataNotes: PropTypes.array.isRequired,
  onDelete: PropTypes.func.isRequired,
  onShowModal: PropTypes.func.isRequired
};

ListItem.defaultProps = {
  dataNotes: [],
  onDelete: noop,
  onShowModal: noop
};