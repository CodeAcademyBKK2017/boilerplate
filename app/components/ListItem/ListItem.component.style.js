import React from 'react';
import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  flatListStyle: {
    height: '30%',
    paddingLeft: 10,
    paddingRight: 10
  },
  divide: {
    height: 1,
    width: '100%',
    backgroundColor: '#E0E0E0'
  },
  marginStyle: {
    marginTop: 10,
    marginBottom: 10
  },
  textTitleStyle: {
    fontWeight: 'bold',
    fontSize: 16
  },
  textContentStyle: {
    color: '#333333'
  },
  modalStyle: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(37, 8, 10, 0.78)'
  }
});