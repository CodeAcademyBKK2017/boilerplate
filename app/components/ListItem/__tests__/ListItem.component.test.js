import 'react-native';
import ListItem from '../ListItem.component';
import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

describe('ListItem', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ListItem />
    );
    expect(tree).toBeDefined();
  });

  it('ListItem: MatchSnapshot', () => {
    const tree = renderer.create(<ListItem />);
    expect(tree).toMatchSnapshot();
  });

  it('ListItem: renderItemList', () => {
    const props = {};
    const wrapper = shallow(<ListItem {...props}/>);
    const instance = wrapper.instance();
    const param = {item: {content: 'Qweqw', key: 0, title: 'Eqweqwe'}};
    const tree = instance.renderItemList(param);
    expect(tree).toMatchSnapshot();
  });
});