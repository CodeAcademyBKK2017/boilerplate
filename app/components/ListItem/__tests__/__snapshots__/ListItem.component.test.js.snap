// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem ListItem: MatchSnapshot 1`] = `
<View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "height": "30%",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`ListItem ListItem: onShowDetail 1`] = `[Function]`;

exports[`ListItem ListItem: renderItemList 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <View
    style={
      Object {
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Eqweqwe
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#333333",
        }
      }
    >
      Qweqw
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#E0E0E0",
        "height": 1,
        "width": "100%",
      }
    }
  />
</TouchableOpacity>
`;
